test/CZMQToolkitTests.h

```diff
@@ -33,64 +30,48 @@ class CZMQToolkitTests : public /*::testing::Test*/ ZeroMQTests {
       std::stringstream makeATarget;
       makeATarget << "ipc:///tmp/ipc.test" << boost::this_thread::get_id();
       mTarget = makeATarget.str();
-      srandom((unsigned) time(NULL));
+      srandom((unsigned) time(nullptr));
       zctx_interrupted = false;
+      if (mContext) {
+         zctx_destroy(&mContext);
+      }
       mContext = zctx_new();
       zctx_set_linger(mContext, 10); // linger for a millisecond on close
       zctx_set_rcvhwm(mContext, 10);
-      zctx_set_sndhwm(mContext, 10);// HWM on internal thread communicaiton
+      zctx_set_sndhwm(mContext, 10); // HWM on internal thread communicaiton
       zctx_set_iothreads(mContext, 2);
-      void* socket = zsocket_new(mContext, ZMQ_REP);
-      if (socket != NULL) {
-         zsocket_set_hwm(socket, 1);
-         zsocket_set_linger(socket, 0);
-         int connectRetries = 100;
-         while ((zsocket_bind(socket, mTarget.c_str()) < 0) && connectRetries-- > 0) {
-            zclock_sleep(100);
-         }
-         if (connectRetries > 0) {
-            mRepSocket = socket;
-            socket = zsocket_new(mContext, ZMQ_REQ);
-            if (socket != NULL) {
-               zsocket_set_hwm(socket, 1);
-               zsocket_set_linger(socket, 0);
-               int connectRetries = 100;
-               while ((zsocket_connect(socket, mTarget.c_str()) < 0) && connectRetries-- > 0) {
-                  zclock_sleep(100);
-               }
-               if (connectRetries > 0) {
-                  mReqSocket = socket;
-               }
-            }
-            socket = zsocket_new(mContext, ZMQ_REQ);
-            if (socket != NULL) {
-               zsocket_set_hwm(socket, 1);
-               zsocket_set_linger(socket, 0);
-               int connectRetries = 100;
-               while ((zsocket_connect(socket, mTarget.c_str()) < 0) && connectRetries-- > 0) {
-                  zclock_sleep(100);
-               }
-               if (connectRetries > 0) {
-                  mReqSocket2 = socket;
-               }
-            }
-         }
-      }
 
+      mRepSocket = zsocket_new(mContext, ZMQ_REP);
+      ASSERT_NE(nullptr, mRepSocket);
+      zsocket_set_hwm(mRepSocket, 1);
+      zsocket_set_linger(mRepSocket, 0);
+      int bindRetries = 100;
+      while ((zsocket_bind(mRepSocket, mTarget.c_str()) < 0) && bindRetries-- > 0) {
+         zclock_sleep(100);
+      }
+      ASSERT_LT(0, bindRetries);
 
+      mReqSocket = zsocket_new(mContext, ZMQ_REQ);
```

Comment:
this reads like we are creating a reply socket, then binding it to the target, and using the same socket pointer to do the same binding for a request socket.

unit tests later use this mReqSocket, does it matter to them whether it is pointing to the request or reply socket?